lib/xandra/protocol.ex

```diff
@@ -26,6 +26,19 @@ defmodule Xandra.Protocol do
     end
   end
 
+  defmacrop decode_string_or_atom({:<-, _, [value, buffer]}, atom_keys?) do
```

Comment:
I believe this macro breaks the single match context optimization.
We should rather conditionally call `String.to_atom/1` after `decode_string/1`.

Reply:
I compiled with bin_opt_info and got:
```
warning: OPTIMIZED: creation of sub binary delayed
  lib/xandra/protocol.ex:811
```

Reply:
On second thought, since there are only two calls, it probably makes sense to get rid of this macro.

Reply:
Indeed, it is optimized. However the produced assembly is longer and less performant than the one we would have with `String.to_atom/1` outside `decode_string/1`.